In [33]:
# imports

import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

%matplotlib inline

# Data Exploration

In [34]:
# reading the dataset

data = pd.read_csv(r'C:\Github\Real-Estate-Price-Prediction\Real-Estate-Price-Prediction\Data\house_prices.csv')

In [35]:
# looking at the data

display(data.head(5),data.sample(5),data.tail(5))

area_type   availability                  location       size  \
0  Super built-up  Area         19-Dec  Electronic City Phase II      2 BHK   
1            Plot  Area  Ready To Move          Chikka Tirupathi  4 Bedroom   
2        Built-up  Area  Ready To Move               Uttarahalli      3 BHK   
3  Super built-up  Area  Ready To Move        Lingadheeranahalli      3 BHK   
4  Super built-up  Area  Ready To Move                  Kothanur      2 BHK   

   society total_sqft  bath  balcony   price  
0  Coomee        1056   2.0      1.0   39.07  
1  Theanmp       2600   5.0      3.0  120.00  
2      NaN       1440   2.0      3.0   62.00  
3  Soiewre       1521   3.0      1.0   95.00  
4      NaN       1200   2.0      1.0   51.00

area_type   availability         location       size  \
3899   Super built-up  Area  Ready To Move       HBR Layout      2 BHK   
11868  Super built-up  Area         18-Mar         Kothanur      2 BHK   
8200         Built-up  Area  Ready To Move        EPIP Zone      3 BHK   
2270             Plot  Area  Ready To Move        TC Palaya  4 Bedroom   
3487         Built-up  Area         18-Dec  Electronic City      2 BHK   

       society total_sqft  bath  balcony   price  
3899   Flretes       1200   2.0      2.0   60.00  
11868      NaN       1185   2.0      2.0   59.00  
8200       NaN       2500   3.0      1.0  268.00  
2270       NaN       4800   3.0      3.0  302.00  
3487       NaN       1020   2.0      1.0   29.46

area_type   availability               location       size  \
13315        Built-up  Area  Ready To Move             Whitefield  5 Bedroom   
13316  Super built-up  Area  Ready To Move          Richards Town      4 BHK   
13317        Built-up  Area  Ready To Move  Raja Rajeshwari Nagar      2 BHK   
13318  Super built-up  Area         18-Jun        Padmanabhanagar      4 BHK   
13319  Super built-up  Area  Ready To Move           Doddathoguru      1 BHK   

       society total_sqft  bath  balcony  price  
13315  ArsiaEx       3453   4.0      0.0  231.0  
13316      NaN       3600   5.0      NaN  400.0  
13317  Mahla T       1141   2.0      1.0   60.0  
13318  SollyCl       4689   4.0      1.0  488.0  
13319      NaN        550   1.0      1.0   17.0

In [37]:
# checking the data summary
data.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 13320 entries, 0 to 13319
Data columns (total 9 columns):
 #   Column        Non-Null Count  Dtype  
---  ------        --------------  -----  
 0   area_type     13320 non-null  object 
 1   availability  13320 non-null  object 
 2   location      13319 non-null  object 
 3   size          13304 non-null  object 
 4   society       7818 non-null   object 
 5   total_sqft    13320 non-null  object 
 6   bath          13247 non-null  float64
 7   balcony       12711 non-null  float64
 8   price         13320 non-null  float64
dtypes: float64(3), object(6)
memory usage: 936.7+ KB


**Inference:**

- there are 13320 records with 8 columns.
- total_sqft, bath, balcony, price are the float64 values rest are of object data type.
- there seem to have misssing values in the data.

In [39]:
# checking the central tencendy of the data

df.describe().T

count        mean         std  min   25%   50%    75%     max
bath     13247.0    2.692610    1.341458  1.0   2.0   2.0    3.0    40.0
balcony  12711.0    1.584376    0.817263  0.0   1.0   2.0    2.0     3.0
price    13320.0  112.565627  148.971674  8.0  50.0  72.0  120.0  3600.0

**Inference:**

- There is a posibility of outliers as there is property with 40 number of baths which is extremely rare. Every property has minimum 1 bathrom with average of 2 bathrooms per property.
- There number of balcony varies from minimum 0 to maximum 3.
- The price column is not interpretable at the moment.